packages/core/src/app/shipping/PayPalFastlaneShippingAddress.tsx

```diff
@@ -22,20 +50,110 @@ export const PayPalFastlaneShippingAddress: FC<PayPalFastlaneShippingAddressProp
         deinitialize,
         isLoading,
         shippingAddress,
+        addresses,
+        isGuest,
+        handleFieldChange,
+        consignments
     } = props;
+    const paypalFastlaneShippingComponent = useRef<PayPalFastlaneAddressComponentRef>({});
+    const paypalCommerceFastlaneOptions = {
+        paypalcommercefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
 
-    return (
-        <PayPalFastlaneShippingAddressForm
-            address={shippingAddress}
-            countries={countries}
-            deinitialize={deinitialize}
-            formFields={formFields}
-            initialize={initialize}
-            isLoading={isLoading}
-            methodId={methodId}
-            onAddressSelect={onAddressSelect}
-            onFieldChange={onFieldChange}
-            onUnhandledError={onUnhandledError}
-        />
+    const braintreeFastlaneOptions = {
+        braintreefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
+
+    const initializationOptions: ShippingInitializeOptions = isPayPalCommerceFastlaneMethod(
+        methodId,
     )
+        ? paypalCommerceFastlaneOptions
+        : braintreeFastlaneOptions;
+
+    const initializeShippingStrategyOrThrow = async () => {
+        try {
+            await initialize({
+                methodId,
+                ...initializationOptions,
+            });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    const deinitializeShippingStrategyOrThrow = async () => {
+        try {
+            await deinitialize({ methodId });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    useEffect(() => {
+        void initializeShippingStrategyOrThrow();
+
+        return () => {
+            void deinitializeShippingStrategyOrThrow();
+        };
+    }, []);
+
+    const { isPayPalFastlaneEnabled, paypalFastlaneAddresses, shouldShowPayPalFastlaneShippingForm } = usePayPalFastlaneAddress();
+
+    const shippingAddresses = isPayPalFastlaneEnabled && isGuest
```

Comment:
Why do we need to provide paypalFastlaneAddresses to default ShippingAddressForm for guests? 

Shopper may have access to paypalFastlaneAddresses only when the Shopper is a Fastlane user. Therefore PayPalFastlaneShippingAddressForm should be rendered for such shoppers

What may happen when shippingAddresses variable will be removed?

Reply:
fixed